### DISCUSSION TITLE
Auth-refresh returns the same token string? I probably do not understand how auth works

### DISCUSSION THREAD

**Discussion by Gys** [pocketbase/pocketbase#5608] [Q&A] [answered]
Hi

I have a Pocketbase based Go backend. Users have a mobile app in RN that access that backend. A user registers and the backend sends the user an auth token generated using `tokens.NewRecordAuthToken(pb, <user>)`.

I understand the auth token is only valid for 14 days (default setting in PB). So the auth token needs to be refreshed regularly. For that the user / mobile app makes a POST request to `/api/collections/users/auth-refresh` (with header `Authorization: <auth token>`). This request returns an object with user info plus an auth token.

So far so good.

However, noticed the 'new' auth token returned by this request fully matches the 'previous' auth token? I would expect to receive a new auth token which invalidates any previous token. So did I do something wrong or is this by design? Does that mean some timer is reset somewhere to restart the 14 days validation?

**Comment by ganigeorgiev** [comment, marked as answer]
`auth-refresh` never invalidates previous tokens. It simply returns the latest user data + creates a new token with refreshed `exp` claim. By default PocketBase APIs are fully stateless and the generated tokens are not stored on the server.

Regarding the new auth token matching the old one - the only cases I think this could happen could be:
- there is a cache on reverse proxy level (if any)
- the 2 tokens were generated with < 1 second difference (the `exp` claim format is in unix seconds)

**Reply by Gys** [reply]
Thank you for the very quick response!

> the 2 tokens were generated with < 1 second difference (the exp claim format is in unix seconds)

Correct, I did not mention the equal tokens were generated during tests, right after each other. I did not consider that timing might be a thing. Indeed adding a one second delay makes the strings different.
